runtime/libc/uvwasi_backing.c

```diff
@@ -0,0 +1,1403 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <uvwasi.h>
+
+#include "../runtime.h"
+
+/* Code that actually runs the wasm code */
+IMPORT void wasmf__start(void);
+
+#define unlikely(x) __builtin_expect(!!(x), 0)
+
+static uvwasi_t uvwasi;
+
+/* TODO: This likely makes assumptions about the memory model and needs to be generalized */
+static inline void check_bounds(uint32_t offset, uint32_t bounds_check)
+{
+    // Due to how we setup memory for x86, the virtual memory mechanism will catch the error, if bounds < WASM_PAGE_SIZE
+    assert(bounds_check < WASM_PAGE_SIZE || (memory_size > bounds_check && offset <= memory_size - bounds_check));
+}
+
+/**
+ * @brief Lifecycle function that runs at end at termination of WebAssembly module. Suitable for logging and cleanup
+ */
+void runtime_on_module_exit()
+{
```

Comment:
The formatting in this file doesn't match the rest of the runtime. I think there is an auto format file in the repo

Reply:
I _think_ that may only be in PR #31. I rebased that and marked for review. Happy to rebase and reformat that here once that's approved and merged.